lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3294,18 +3294,52 @@ private function performCallbackOnCachedPersister(callable $callback)
         }
     }
 
+    private function dispatchFlushEntityListener($event)
+    {
+        $subscribedSystems = [];
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
+            foreach ($this->identityMap as $className => $entities) {
+                $hash = (isset($this->entityIdentifiers[$oid], $this->entityIdentifiers[$oid]['id'])) ? $this->entityIdentifiers[$oid]['id'] : null;
```

Comment:
I may have some time later today or tomorrow to give you a better answer than this one, but the main reason that I was using the nested `foreach` loop was due to an attempt at creating logic similar to that of the other Entity Listeners, but on the `onFlush` and `postFlush` events, which do not digest each entity as easily as the other events, which can piggyback off the loops already available.